Makefile

```diff
@@ -68,6 +70,21 @@ gh-publish:
 	git commit -m "Generated $(GH_PUBLISH_BRANCH) for `git log $(GH_SOURCE_BRANCH) -1 --pretty=short --abbrev-commit`" && git push -f $(GH_UPSTREAM_REPO) $(GH_PUBLISH_BRANCH)
 	git checkout $(GH_SOURCE_BRANCH)
 
+docker-gh-stage:
+	git checkout $(GH_PUBLISH_BRANCH)
+	git checkout $(GH_SOURCE_BRANCH) -- $(GH_SOURCE_DIR)
+	git reset HEAD 
+	make clean
+	make html
+	rsync -a $(BUILDDIR)/* .
+	rsync -a $(BUILDDIR)/.* .
+	git add `(cd $(BUILDDIR); find . -type f; cd ..)`
+	rm -rf $(GH_SOURCE_DIR) $(BUILDDIR)
+
+docker-gh-push:
+	git commit -m "Generated $(GH_PUBLISH_BRANCH) for `git log $(GH_SOURCE_BRANCH) -1 --pretty=short --abbrev-commit`" && git push -f $(GH_UPSTREAM_REPO) $(GH_PUBLISH_BRANCH)
```

Comment:
my only problem here, is that `docker-gh-push` could be run from any branch, while previous `gh-publish` checked out the `$(GH_PUBLISH_BRANCH)`

maybe we could add a token file that would be created with `docker-gh-stage`, and removed with `docker-gh-publish` to ensure that they are run one after an other ?

Reply:
I added a checkout for `GH_PUBLISH_BRANCH` at the beginning of the publish step